File: baekjoon/js/ex02_implementation/11723.js
let s;
let ans = [];

const operation = (str, x) => {
  switch (str) {
    case "add":
      s[Number(x)] = true;
      break;
    case "remove":
      s[Number(x)] = false;
      break;
    case "check":
      ans.push(s[Number(x)] ? 1 : 0);
      break;
    case "toggle":
      s[Number(x)] ? (s[Number(x)] = false) : (s[Number(x)] = true);
      break;
    case "all":
      s = Array.from({ length: 21 }, (v) => true);
      break;
    case "empty":
      s = Array.from({ length: 21 }, (v) => false);
      break;
  }
};

const arr = () => {
  let input = require("fs")
    .readFileSync("/dev/stdin")
    .toString()
    .trim()
    .split("\n")
    .map((v) => v.split(" "));
  s = Array.from({ length: 21 }, (v) => false);
  const m = input[0][0];
  for (let i = 1; i <= m; i++) operation(input[i][0], input[i][1], s, ans);
  console.log(ans.join("\n"));
};

arr();

File: baekjoon/C++/ex02_implementation/11723.cpp
#include <iostream>

int main(void) {
	int m, n, s = 0;
	std::string str;

	std::ios_base::sync_with_stdio(false);
	std::cin.tie(0);
	std::cout.tie(0);

	std::cin >> m;
	for (int i = 0; i < m; i++) {
		std::cin >> str;
		if (str == "add") {
			std::cin >> n;
			s |= (1 << n);
		}
		else if (str == "remove") {
			std::cin >> n;
			s &= ~(1 << n);
		}
		else if (str == "check") {
			std::cin >> n;
			if (s & (1 << n)) std::cout << "1\n";
			else std::cout << "0\n";
		}
		else if (str == "toggle"){
			std::cin >> n;
			s ^= (1 << n);
		}
		else if (str == "all")
			s = (1 << 21) - 1;
		else if (str == "empty")
			s = 0;
	}
}
